### ISSUE TITLE
find fails on nested empty tags

### ISSUE THREAD

**Issue by maxhumber** [maxhumber/gazpacho#7]
**Describe the bug**
The `find` method gets confused on empty element tags (img, meta, etc...) 

**To Reproduce**
Code to reproduce the behaviour:

```python
from gazpacho import Soup

html = '''
<div class="foo-list">
  <a class="foo" href="/foo/1">
    <div class="foo-image-container">
      <img src="image.jpg">
    </div>
  </a>
  <a class="foo" href="/foo/2">
    <div class="foo-image-container">
      <img src="image.jpg">
    </div>
  </a>
</div>
'''

soup = Soup(html)
soup.find('a', {'class': "foo"})
```

**Expected behavior**
`find` should be able to "find" a list of two `a` tags. Instead the full blob is getting returned. 

**Environment:**
 - OS: macOS
 - Version: 0.8

**Comment by maxhumber** [comment]
List of empty element tags: 

- area
- base
- br
- col
- embed
- hr
- img
- input
- keygen
- link
- meta
- param
- source
- track
- wbr

[Source](https://developer.mozilla.org/en-US/docs/Glossary/Empty_element)

**Comment by maxhumber** [comment]
fixed in #8
